Sources/MediaViewer/PreviewItemView/Builtins/PlayerPreviewItemViewController.swift

```diff
@@ -1,51 +1,66 @@
 import UIKit
 import AVKit
+import VideoToolbox
 import os
 
 fileprivate let logger = Logger(
     subsystem: Bundle.main.bundleIdentifier! + ".logger",
     category: #file
 )
 
-public final class PlayerPreviewItemViewController: UIViewController {
-    private let playerView = AVPlayerView()
-    
-    let player: AVPlayer
+@MainActor
+public final class PlayerPreviewItemViewController: AVPlayerViewController {
+    private var playerOutput: AVPlayerItemVideoOutput
     
     public init(player: AVPlayer) {
-        self.player = player
+        playerOutput = AVPlayerItemVideoOutput(pixelBufferAttributes:
+                                                [kCVPixelBufferPixelFormatTypeKey as String: Int(kCVPixelFormatType_32BGRA)])
         super.init(nibName: nil, bundle: nil)
+        self.player = player
+        allowsPictureInPicturePlayback = false
     }
     
     required init?(coder: NSCoder) { fatalError() }
     
-    public override func loadView() {
-        view = playerView
-    }
     
     public override func viewDidLoad() {
         super.viewDidLoad()
-        playerView.playerLayer.player = player
+        player?.currentItem?.add(playerOutput)
     }
     
     public override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
         logger.debug("\(#function)")
-        player.play()
-        navigationController?.topViewController?.toolbarItems = [
-            UIBarButtonItem.seekbar(player)
-        ]
+        player?.play()
+        if let navigationController = navigationController as? NavigationController {
+            navigationController.setBarHidden(false, animated: true)
+        }
     }
     
     public override func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
         logger.debug("\(#function)")
-        player.pause()
-        navigationController?.topViewController?.toolbarItems = []
+        player?.pause()
     }
 }
 
-fileprivate final class AVPlayerView: UIView {
-    override class var layerClass: AnyClass { AVPlayerLayer.self }
-    var playerLayer: AVPlayerLayer { layer as! AVPlayerLayer }
+@MainActor
+extension PlayerPreviewItemViewController: DismissTransitionViewProviding {
+    public var viewForDismissTransition: UIView? {
+        guard let player else { return nil }
+        if let asset = player.currentItem?.asset {
```

Comment:
Shouldn't this be a guard? Does this have any meaning if we don't have an asset or a player?